api/resources_portal/views/attachment.py

```diff
@@ -24,9 +26,16 @@ class Meta:
 
 
 def user_has_perm_on_active_material_request(user, perm):
-    for organization in user.organizations.all():
+
+    # Retrieve all organization permissions in a single query
+    checker = ObjectPermissionChecker(user)
+    organizations = Organization.objects.all()
+    checker.prefetch_perms(organizations)
+
+    # Uses prefetch_related to retrieve all related objects in a single query
+    for organization in user.organizations.all().prefetch_related("materials"):
         if user.has_perm(perm, organization):
```

Comment:
I think you need to use the `checker` here.

Reply:
Ahh, you're right. Good catch.